sql/delete.go

```diff
@@ -163,7 +163,7 @@ func canDeleteWithoutScan(n *parser.Delete, rows planNode, indexCount int) bool
 			}
 			return false
 		}
-		if n.Where != nil {
+		if scan := sel.table.node.(*scanNode); scan.filter != nil {
```

Comment:
well, actually it can't be an `indexJoin` since you can only `DELETE FROM ` a single `relation_expr_opt_alias` -- I have a followup PR that moves these casts up to the top of the `Delete` so they'll fail quickly and loudly if that assumption is ever invalidated.